tools/build_defs/configure_script.bzl

```diff
@@ -34,27 +35,50 @@ def create_configure_script(
     )]
     return "\n".join(script)
 
-def _define_deps_flags(inputs):
-    libs = collect_libs(inputs.deps_linking_info)
-
+def _define_deps_flags(deps, inputs):
     # It is very important to keep the order for the linker => put them into list
     lib_dirs = []
 
-    # For filtering duplicates
+    # Here go libraries built with Bazel
     lib_dirs_set = {}
-    for lib in libs:
+    for lib in inputs.libs:
         dir = lib.dirname
         if not lib_dirs_set.get(dir):
             lib_dirs_set[dir] = 1
-            lib_dirs += [dir]
+            lib_dirs += ["-L$EXT_BUILD_ROOT/" + dir]
+
+    include_dirs_set = {}
+    for list in inputs.include_dirs:
+        for include_dir in list:
+            include_dirs_set[include_dir] = "-I$EXT_BUILD_ROOT/" + include_dir
+    for list in inputs.headers:
+        for header in list:
+            include_dir = header.dirname
+            if not include_dirs_set.get(include_dir):
+                include_dirs_set[include_dir] = "-I$EXT_BUILD_ROOT/" + include_dir
+    include_dirs = include_dirs_set.values()
+
+    # For the external libraries, we need to refer to the places where
+    # we copied the dependencies ($EXT_BUILD_DEPS/<lib_name>), because
+    # we also want configure to find that same files with pkg-config
+    # -config or other mechanics.
+    # Since we need the names of include and lib directories under
+    # the $EXT_BUILD_DEPS/<lib_name>, we ask the provider.
+    lib_dirs_set = {}
+    for dep in deps:
```

Comment:
Same question here, is this not flattening the depset at every level?

Reply:
Yes, this is flattening, needed for script text creation.